### PR TITLE
[TC-7427] Add retry strategies

### PR DESCRIPTION
**Pull request by marcmatt**
## Description

https://tradecloud.atlassian.net/browse/TC-7427

### Scope
This PR adds client retry strategies to the rules.

### Submitter pre-flight check

- [x] Review your documentation
- [x] Add labels `needs reviewing`
- [x] Assign PR and JIRA ticket to 'Reviewer' (in 'Reviewing' state)

### Reviewer pre-flight check

- [ ] Review documentation in PR
- [ ] Remove label `needs reviewing`
- [ ] Merge the PR and remove the release for this branch from Gitbook
- [ ] Set Jira ticket to 'Released'

### CODE REVIEW

File: api/rules.md [hunk 1]
```diff
@@ -54,16 +54,31 @@ Your identifiers **must not contain whitespace characters.**
...
+
+Send the next message when the previous one has been responded with HTTP Status Code 200 OK or 202 Accepted.
 
 Do **not** send more than **10 requests per second**.
 
 Tradecloud may respond with [HTTP Status Code 429](https://tools.ietf.org/html/rfc6585#section-4) `Too Many Requests`
 
+### Queue and retry messages with an exponential backoff.
+
+The Tradecloud [environments](environments.md) are not 100% available. It is the API client's responsibility to queue the message and automatically retry with exponential backoff till Tradecloud is available again. An alternative is to warn the ERP user, who is trying to send the message to Tradecloud, so the user can manually retry later.
```
Comment (roy-tc) [inline api/rules.md:71]:
```suggestion
The Tradecloud [environments](environments.md) do not have an availability SLO of 100%. If Tradecloud is temporarily unavailable, it is the API client's responsibility to queue the message and automatically retry with exponential backoff till Tradecloud is available again. An alternative is to warn the ERP user, who is trying to send the message to Tradecloud, so the user can manually retry later.
```

File: api/rules.md [hunk 2]
```diff
@@ -54,16 +54,31 @@ Your identifiers **must not contain whitespace characters.**
...
+
+When Tradecloud responds with either a: 
+
+- any HTTP Status Code 5xx Server Error or
+- [HTTP Status Code 429](https://tools.ietf.org/html/rfc6585#section-4) `Too Many Requests` or 
+- the requests does time out (currently 5 secs. at Tradecloud API side)
+  
+Then the client must automatically retry using an exponential backoff strategy or use a manual retry strategy. 
+The [Exponential Backoff Calculator](http://backoffcalculator.com/?interval=5&rate=2&attempts=5) is handy to verify the retry plan.
+
```
Comment (roy-tc) [inline api/rules.md:81]:
```suggestion
If Tradecloud responds with either: 

- a HTTP Status Code 5xx Server Error or
- [HTTP Status Code 429](https://tools.ietf.org/html/rfc6585#section-4) `Too Many Requests` or 
- the requests does time out (currently 5 secs. at Tradecloud API side)
  
Then the client must automatically retry, using an exponential backoff strategy, or use a manual retry strategy. 
The [Exponential Backoff Calculator](http://backoffcalculator.com/?interval=5&rate=2&attempts=5) is handy to verify the retry plan.

```

**Review by roy-tc — APPROVED** [review]
@marcmatt looks good. I've suggested some small nits, but otherwise can be merged as is

**Comment by marcmatt** [comment]
@roy-tc Thanks for the suggestions, merged them.

**Pull request merged**